src/integrations/prefect-sqlalchemy/prefect_sqlalchemy/credentials.py

```diff
@@ -103,7 +103,9 @@ class ConnectionComponents(BaseModel):
     driver: Union[AsyncDriver, SyncDriver, str] = Field(
         default=..., description="The driver name to use."
     )
-    database: str = Field(default=..., description="The name of the database to use.")
+    database: Optional[str] = Field(
+        default=None description="The name of the database to use."
```

Comment:
I think there will be some implications associated with this being  optional now
```suggestion
        default=None, description="The name of the database to use."
```